lib/subscription-client.js

```diff
@@ -36,17 +38,20 @@ class SubscriptionClient {
     this.reconnectAttempts = 0
     this.connectionCallback = connectionCallback
     this.failedReconnectCallback = failedReconnectCallback
+    this.connectionInitPayload = connectionInitPayload
 
     this.connect()
   }
 
   connect () {
     this.socket = new WebSocket(this.uri, this.protocols)
 
-    this.socket.onopen = () => {
+    this.socket.onopen = async () => {
```

Comment:
Can you add a try/catch? The `connectionInitPayload` can throw, and it will lead to an unhandledRejection